File: demo/flutternew/lib/domain/dto/BaseResponse.dart
class BaseResponse<T> {
  final String code;
  final String? message;
  final T? data;

  const BaseResponse({
    required this.code,
    this.message,
    this.data,
  });

  bool get isSuccess => code == '200';

  factory BaseResponse.fromJson(
    Map<String, dynamic> json,
    T Function(Object? json) fromJsonT,
  ) {
    return BaseResponse(
      code: json['code']?.toString() ?? '',
      message: json['message']?.toString(),
      data: json['data'] == null ? null : fromJsonT(json['data']),
    );
  }

  Map<String, dynamic> toJson(Object? Function(T value) toJsonT) {
    return {
      'code': code,
      'message': message,
      'data': data == null ? null : toJsonT(data as T),
    };
  }
}

File: app/android/core/baseutil/src/main/java/com/core/baseutil/network/BaseResponse.kt
package com.core.baseutil.network

import com.core.baseutil.json.GsonBean
import java.io.Serializable

class BaseResponse<T> : GsonBean, Serializable {
    // 泛型字段，用于存放具体的数据
    var code: String? = null
    var message: String? = null
    var data: T? = null
}